doc/usage/restructuredtext/basics.rst

```diff
@@ -605,6 +605,6 @@ There are some problems one commonly runs into while authoring reST documents:
 
 .. rubric:: Footnotes
 
-.. [1] When the default domain contains a :rst:dir:`class` directive, this
-       directive will be shadowed.  Therefore, Sphinx re-exports it as
+.. [1] Since some domains contain a ``class`` directive, it shadows
```

Comment:
You're right: I was confused because `rst-class` is registered always, along with `toctree` and others:

https://github.com/sphinx-doc/sphinx/blob/3d8fbd992cedf1784b106cf43ecbc452953ebe17/sphinx/directives/other.py#L386

But the current wording says "this directive will be shadowed". I see why the change is not good.